### DISCUSSION TITLE
A complete electrode model for plasma impedance probes

### DISCUSSION THREAD

**Discussion by gatling-nrl** [kinnala/scikit-fem#1166] [Show and tell]
Here's a paper I wrote for Physics of Plasmas using scikit-fem. Just thought I'd share. And thanks for the help along the way!

https://doi.org/10.1063/5.0199553

**Comment by kinnala** [comment]
Congrats!

If sky was the limit, what would you have differently in the library?

**Reply by gatling-nrl** [reply]
I would make assembly optionally parallel (parmetis or something?) and make it easier to optionally use petsc as a solver. 

But I really *really* like the simplicity of the numpy/spsolve that skfem has now. Using numpy and scipy as a base made it much easier to learn both fem and skfem,  and to understand what was going on in the codes.

But as the problem scale grew, spsolve was unable to cope, and it was hard to get good performance out of the cluster I was running on.

**Reply by kinnala** [reply]
Were you able to use petsc to solve larger problems? I once tried to use its AMG preconditioner with PCG on a single laptop and was already able to solve roughly 1 magnitude larger 3D problems (in terms of DOFs) than spsolve.

Do you mean something like:
- Out of the box distributed mesh data structure implemented with metis/parmetis and mpi4py
- Support for petsc4py in asm to assemble a distributed petsc matrix over the distributed mesh in parallel

Thanks for the feedback. I will definitely want to experiment more with petsc4py sometime.
